Coding Style Guidline.rst

Stockfoot opened the pull request:
Created a unified version of the Coding Style Guidelines
Extracted rules from various pandas website information as well as submission scripts that verify style requirements
Work in progress

- [x] closes #33851
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

Stockfoot committed:
Coding Style Guidline.rst

Created a unified version of the Coding Style Guidelines
Extracted rules from various pandas website information as well as submission scripts that verify style requirements
Work in progress

WillAyd commented:
This just duplicates what is already on the docs right? If so I think can just update the existing contributing guide with links rather than a new file with copied content

Stockfoot commented:
> 
> 
> This just duplicates what is already on the docs right? If so I think can just update the existing contributing guide with links rather than a new file with copied content

This doesn't just copy what is in the docs. 

This is just the start of the doc and I did pull information scattered in multiple locations about coding guidelines. I also started to pull out checks that are done in the script that check for coding guidelines that are not described anywhere. 

If this document becomes more robust it think that it would be a useful tool for the OSS project.

Stockfoot commented:
I am also unsure of how to fix the CI/Checks error. It is saying there is a trailing white space after all my lines which do no show up in my IDE

WillAyd commented on Coding Style Guidline.rst:
```diff
@@ -0,0 +1,285 @@
+2.1 Line Length
+~~~~~~~~~~~~~~~
+
+Line length is restricted to 80 characters to promote readability.
```
We use a line length of 88

WillAyd commented on Coding Style Guidline.rst:
```diff
@@ -0,0 +1,285 @@
+
+Line length is restricted to 80 characters to promote readability.
+
+2.2 Indentation
```
I think this is enforced by black; no need to detail here

WillAyd commented on Coding Style Guidline.rst:
```diff
@@ -0,0 +1,285 @@
+    if x == 3
+     x = 10
+
+2.3 Whitespaces Around Arithmetic Operators
```
same for comment around black

WillAyd commented on Coding Style Guidline.rst:
```diff
@@ -0,0 +1,285 @@
+
+   x=3+5/2
+
+2.4 Missing White Spaces After Commas
```
same for black

WillAyd commented on Coding Style Guidline.rst:
```diff
@@ -0,0 +1,285 @@
+
+   myFunctionCall('a','b','c')
+
+2.5 Header File
```
Is this a Python reference?

WillAyd commented on Coding Style Guidline.rst:
```diff
@@ -0,0 +1,285 @@
+   value = str
+   f"Unknown recived type, got: '{type(value).__name__}'"
+
+4 Types
```
OK so this is already in the contributing guideline. If we want to move here then should remove from that so as not to duplicate

WillAyd reviewed (commented):
OK thanks. So for things that are enforced by black I don't think we need to go into details as the tool just does these things for you.

We still need to figure out how to manage this versus existing documentation we have, but if you can trim down by removing the black sections should make it easier to iterate

Stockfoot commented:
@WillAyd I'll trim those out and resubmit, thank you for your help!

Stockfoot committed:
Removed Sections Enforced by Black

Stockfoot commented:
@WillAyd I have updated those sections. Let me know if there is anything else that I could research/add

Stockfoot commented:
@WillAyd I am also unsure of how to get past the CI / Checks portion of the pull request. It is saying there is a trailing white space after every line but in my IDE this is not there.

WillAyd commented on Coding Style Guidline.rst:
```diff
@@ -0,0 +1,223 @@
+
+**Good:**
+
+::
```
Can you make these code-block:: python directives?

WillAyd commented on Coding Style Guidline.rst:
```diff
@@ -0,0 +1,223 @@
+2 Patterns
+----------
+
+2.1 Header File
```
What is this in reference to? This isn't anything Python related right? I think should just remove

WillAyd commented on Coding Style Guidline.rst:
```diff
@@ -0,0 +1,223 @@
+   import numpy as np
+   import pandas as pd
+
+4.1.2 Unused Imports
```
Since this is enforced by flake8 already I don't think need to call out here

WillAyd commented on Coding Style Guidline.rst:
```diff
@@ -0,0 +1,223 @@
+
+   maybe_primes: List[Union[int, None]] = []
+
+4.1.1 Redundant Imports
```
This is strictly for writing documentation, but for general code style it is certainly OK to import these. I think can just remove this from this guide

Stockfoot replied:
@WillAyd I made those changes you requested, but it still seems like it is failing after ever carriage line return

WillAyd reviewed (changes requested):
I think the trailing whitespace might be a false positive from not having the `code-block` directives explicitly declared

Stockfoot committed:
Removed Sections and Changed Code Block Directives

jbrockmendel commented:
The section on `repr` can probably go, since we mostly use fstrings now.

WillAyd commented:
@Stockfoot still active? If so can you simplify and address comments?

Stockfoot commented:
@WillAyd I am still around.
I am just unsure of how to proceed. The CI/Check seems to fail no matter what I do.
I have made all the changes that were previously suggested. The only one I need to address is the most recent with @jbrockmendel 
> The section on repr can probably go, since we mostly use fstrings now.

arw2019 reviewed (commented):
@Stockfoot are you interested in continuing? If yes if you can address comments & merge master we'll take another look

Stockfoot commented:
@arw2019 I am interested in continuing. I have addressed the comments but no one has ever responded back to my question on how to proceed and actually get a pull request going. 

Every time I try and submit it will no pass CI / Checks and am unsure how to proceed...

![image](https://user-images.githubusercontent.com/36280534/98286417-37007e00-1f61-11eb-9bc5-4636207c2abc.png)

arw2019 commented:
Can you merge master again and I'll take a look (output from the last build is no longer available for inspection). Some failures are random but we want to make sure otherwise will have to address

Also I might be wrong but I don't think you responded to https://github.com/pandas-dev/pandas/pull/34608/files#r440389770

arw2019 commented:
Closing for now. @Stockfoot let us know whenever you'd like to pick this up and we'll reopen
